### ISSUE TITLE
visibility modifiers are accepted for enum variants

### ISSUE THREAD

**Issue by matklad** [rust-lang/rust#28433]
The following code compiles without warnings:

```rust
#[allow(dead_code)]
enum Foo {
    pub Spam,
    Eggs
}

fn main() {

}
```

I guess that this should be forbidden, because enum vairants inherit visibility of the enum itself. At minimum, this should produce a warning.

This is the relevant line from parser: https://github.com/rust-lang/rust/blob/master/src/libsyntax/parse/parser.rs#L5203

This is the discussion on Rust forum: https://users.rust-lang.org/t/confused-about-enum-variants-visibility/2873

**Comment by matklad** [comment]
The following tests fail (due to different error message), if variants visibility modifiers are forbidden:

```
test [compile-fail] compile-fail/issue-3993-2.rs ... FAILED
test [compile-fail] compile-fail/useless-priv.rs ... FAILED
```

**Comment by nagisa** [comment]
I’m fine with making this change without an RFC, but we should still run a crater run with change applied.
